src/Symfony/Component/Form/ChoiceList/Factory/ChoiceListFactoryInterface.php

```diff
@@ -80,18 +80,20 @@ public function createListFromLoader(ChoiceLoaderInterface $loader, $value = nul
      * match the keys of the choices. The values should be arrays of HTML
      * attributes that should be added to the respective choice.
      *
-     * @param ChoiceListInterface $list             The choice list
-     * @param array|callable|null $preferredChoices The preferred choices
-     * @param callable|null       $label            The callable generating the
-     *                                              choice labels
-     * @param callable|null       $index            The callable generating the
-     *                                              view indices
-     * @param callable|null       $groupBy          The callable generating the
-     *                                              group names
-     * @param array|callable|null $attr             The callable generating the
-     *                                              HTML attributes
+     * @param ChoiceListInterface $list                      The choice list
+     * @param array|callable|null $preferredChoices          The preferred choices
+     * @param callable|null       $label                     The callable generating the
+     *                                                       choice labels
+     * @param callable|null       $index                     The callable generating the
+     *                                                       view indices
+     * @param callable|null       $groupBy                   The callable generating the
+     *                                                       group names
+     * @param array|callable|null $attr                      The callable generating the
+     *                                                       HTML attributes
+     * @param bool                $keepPreferredChoicesOrder Whether to keep preferred
+     *                                                       choices order or not
      *
      * @return ChoiceListView The choice list view
      */
-    public function createView(ChoiceListInterface $list, $preferredChoices = null, $label = null, $index = null, $groupBy = null, $attr = null);
+    public function createView(ChoiceListInterface $list, $preferredChoices = null, $label = null, $index = null, $groupBy = null, $attr = null/*, bool $keepPreferredChoicesOrder = false */);
```

Comment:
What about making a default `true` in 5.0? This would imply deprecating passing `false` for BC.

Reply:
this in turn would mean to require passing the argument (because that's the only possible BC layer here)